src/js/providers/default.ts

```diff
@@ -0,0 +1,238 @@
+import { PLAYER_STATE, MEDIA_TYPE } from 'events/events';
+import type { GenericObject, SourceObj, SeekRange } from 'types/generic.type';
+import type { TracksMixin, SimpleAudioTrack } from 'providers/tracks-mixin';
+import type { VideoActionsInt } from 'providers/video-actions-mixin';
+import type { VideoAttachedInt } from 'providers/video-attached-mixin';
+import type Events from 'utils/backbone.events';
+import type PlaylistItem from 'playlist/item';
+
+const noop: () => void = function(): void { /* noop */ };
+const returnFalse: () => boolean = (() => false);
+const getNameResult: { name: string } = { name: 'default' };
+const returnName: () => { name: string } = (() => getNameResult);
+
+export interface ImplementedProvider {
```

Comment:
would love your thoughts here @robwalch  on things we should mandate for external provider integration vs things defined on our internal mixins (resize is a great example - it's defined in the mixin so it does not need to be defined given the union type)

Reply:
`ProviderWithMixins` and `ImplementedProvider` (or `ProviderBaseDependantOnMixins`) should be defined in another file, so that `DefaultProvider` only defines the interface required of any provider.

Reply:
@robwalch , what do you suggest as far as a diff location? I think having it in default shows the additional arguments that `DefaultProvider` does not enumerate. It also gives indications of what we do on our end to integrate with the base interface which I can see as helpful.

Thoughts?

Reply:
I think the mixins are specific to video element playback and TextTrack handling in our providers, and not relevant to anyone building a provider externally unless they also import the mixins from open source. The mixins may work together and with one of our providers but they should be in no way shape or form relied upon in the player of any provider.

Reply:
The point is to separate what public methods of a provider are used outside of the providers/ modules by the player, from the methods that could be private or protected which are specific to our own provider implementations.

- Types for event objects emitted by providers? Sure put those in DefaultProvider
- Mixin methods used by other mixins or html5 provider? Nobody needs that but us.

Reply:
yeah this is all great feedback, updating with an `ImplementedProvider` which exists as our externally facing provider interface, and building `ProviderWithMixins` for internal use (should be castable back and forth given common interface)